client/host.go

```diff
@@ -26,30 +37,81 @@ func (c *Client) GetHostByName(nameLabel string) (hosts []Host, err error) {
 	return obj.([]Host), nil
 }
 
-func FindHostForTests(host *Host) {
-	hostName, found := os.LookupEnv("XOA_HOST")
+func (c *Client) GetHostById(id string) (host Host, err error) {
+	obj, err := c.FindFromGetAllObjects(Host{Id: id})
+	if err != nil {
+		return
+	}
+	hosts, ok := obj.([]Host)
+
+	if !ok {
+		return host, errors.New("failed to coerce response into Host slice")
+	}
 
-	if !found {
-		fmt.Println("The XOA_HOST environment variable must be set")
-		os.Exit(-1)
+	if len(hosts) != 1 {
+		return host, errors.New(fmt.Sprintf("expected a single host to be returned, instead received: %d in the response: %v", len(hosts), obj))
 	}
+
+	return hosts[0], nil
+}
+
+func FindHostForTests(hostId string, host *Host) {
 	c, err := NewClient(GetConfigFromEnv())
 	if err != nil {
 		fmt.Printf("failed to create client with error: %v", err)
 		os.Exit(-1)
 	}
 
-	hosts, err := c.GetHostByName(hostName)
+	queriedHost, err := c.GetHostById(hostId)
 
 	if err != nil {
-		fmt.Printf("failed to find a hosts with name: %v with error: %v\n", hostName, err)
+		fmt.Printf("failed to find a host with id: %v with error: %v\n", hostId, err)
 		os.Exit(-1)
 	}
 
-	if len(hosts) != 1 {
-		fmt.Printf("Found %d hosts with name_label %s. Please use a label that is unique so tests are reproducible.\n", len(hosts), hostName)
-		os.Exit(-1)
+	*host = queriedHost
+}
+
+func (c *Client) GetHostsByPoolName(pool string, tags ...interface{}) (hosts []map[string]interface{}, err error) {
+	hosts = make([]map[string]interface{}, 0)
+	obj, err := c.FindFromGetAllObjects(Host{Pool: pool, Tags: tags})
+
+	if err != nil {
+		return
 	}
+	slice := obj.([]Host)
+	for _, v := range slice {
+		hosts = append(hosts, convertToMap(v))
+	}
+	return hosts, nil
+}
 
-	*host = hosts[0]
+func convertToMap(st interface{}) map[string]interface{} {
```

Comment:
I don't think this reflection complexity is necessary. In the cases where it enables more powerful APIs (such as in the `FindFromGetAllObjects` case), I think it's worth it. However, I think we should just write a function to convert a `Host` struct to a map or we could use a library (like [mapstructure](https://github.com/mitchellh/mapstructure)) that can accomplish this for us (so we don't need to reinvent the wheel for each XO object type this client supports).

I'm fine with either approach. `mapstructure` would reduce the struct <=> map work in other places in the provider but I just haven't made it a priority to fix yet.